metacat-main/src/main/java/com/netflix/metacat/main/services/notifications/sns/SNSNotificationServiceUtil.java

```diff
@@ -0,0 +1,189 @@
+     * @return TablePartitionsUpdatePayload
+     */
+    public TablePartitionsUpdatePayload createTablePartitionsUpdatePayload(
+        final List<PartitionDto> partitionDtos,
+        final MetacatEvent event) {
+        final List<String> deleteColumnValues;
+        String latestDeleteColumnValue = null;
+        final Optional<ObjectNode> objectNode = this.userMetadataService.getDefinitionMetadata(
+            QualifiedName.ofTable(event.getName().getCatalogName(), event.getName().getDatabaseName(),
+                event.getName().getTableName()));
+        String message = ALL_FUTURE_PARTITION_KEY;
+        if (objectNode.isPresent()
+            && !objectNode.get().at(DELETION_COLUMN_PATH).isMissingNode()
+            && !objectNode.get().at(PARTITION_COLUMN_DATA_TYPE_PATH).isMissingNode()) {
+            final String deleteColumn = objectNode.get().at(DELETION_COLUMN_PATH).textValue();
+            deleteColumnValues = getSortedDeletionPartitionKeys(partitionDtos, deleteColumn);
+            //using utc now as today
+            final long nowSecond = Instant.now().getEpochSecond();
+            //convert the value to utc then compare
+            for (String val : deleteColumnValues) {
+                try {
+                    final Long timestamp = getTimeStamp(val,
+                        PST_TIME.contains(objectNode.get().at(PARTITION_COLUMN_DATA_TYPE_PATH).textValue()));
+                    if (timestamp <= nowSecond) {
+                        latestDeleteColumnValue = deleteColumn + "=" + val; //the delete column with value
+                        message = FOUND_VALID_PARTITION_KEY;
+                        break;
+                    }
+                } catch (ParseException ex) {
+                    message = INVALID_PARTITION_KEY_FORMAT;
+                    log.debug("Failure of getting latest key due to invalid timestamp format {} {}:{}",
+                        event.getName().getTableName(), deleteColumn, val);
+                    break;
+                }
+            }
+        } else {
+            message = NOT_ENOUGH_METADATA_INFO;
+        }
+        return new TablePartitionsUpdatePayload(
+            latestDeleteColumnValue,
+            partitionDtos.size(),
+            0,
+            message
+        );
+    }
+
+    /**
+     * get descending order deletion column value.
+     *
+     * @param partitionDtos partition DTOs
+     * @param deleteColumn  delete column name
+     * @return descending order deletion column
+     */
+    public List<String> getSortedDeletionPartitionKeys(final List<PartitionDto> partitionDtos,
+                                                       final String deleteColumn) {
+        List<String> ret = partitionDtos.stream().map(
+            x -> {
+                final List<String> parts = Arrays.asList(x.getName().toString().split("/"));
+                String deletePartName = "";
+                for (String part : parts) {
```

Comment:
You could use com.netflix.metacat.common.server.partition.util.PartitionUtil to get the partition key values as a Map.

Reply:
good to know.